src/client/datascience/interactive-common/notebookServerProvider.ts

```diff
@@ -36,38 +33,34 @@ export class NotebookServerProvider implements IJupyterServerProvider {
         @inject(ProgressReporter) private readonly progressReporter: ProgressReporter,
         @inject(IConfigurationService) private readonly configuration: IConfigurationService,
         @inject(IJupyterExecution) private readonly jupyterExecution: IJupyterExecution,
-        @inject(IApplicationShell) private readonly applicationShell: IApplicationShell,
         @inject(IInterpreterService) private readonly interpreterService: IInterpreterService,
         @inject(IJupyterServerUriStorage) private readonly serverUriStorage: IJupyterServerUriStorage,
-        @inject(JupyterServerSelector) private serverSelector: JupyterServerSelector
+        @inject(JupyterServerSelector) private serverSelector: JupyterServerSelector,
+        @inject(IDisposableRegistry) private readonly disposables: IDisposableRegistry
     ) {}
-    public async getOrCreateServer(
-        options: GetServerOptions,
-        token?: CancellationToken
-    ): Promise<INotebookServer | undefined> {
+    public async getOrCreateServer(options: GetServerOptions): Promise<INotebookServer | undefined> {
         const serverOptions = await this.getNotebookServerOptions(options.resource);
 
         // If we are just fetching or only want to create for local, see if exists
         if (options.getOnly || (options.localOnly && !serverOptions.uri)) {
             return this.jupyterExecution.getServer(serverOptions);
         } else {
             // Otherwise create a new server
-            return this.createServer(options, token);
+            return this.createServer(options);
         }
     }
 
-    private async createServer(
-        options: GetServerOptions,
-        token?: CancellationToken
-    ): Promise<INotebookServer | undefined> {
+    private async createServer(options: GetServerOptions): Promise<INotebookServer | undefined> {
         // When we finally try to create a server, update our flag indicating if we're going to allow UI or not. This
         // allows the server to be attempted without a UI, but a future request can come in and use the same startup
-        this.ui.disableUI = options.ui.disableUI;
-        options.ui.onDidChangeDisableUI(() => (this.ui.disableUI = options.ui.disableUI));
+        if (!options.ui.disableUI) {
+            this.ui.disableUI = false;
+        }
+        options.ui.onDidChangeDisableUI(() => (this.ui.disableUI = options.ui.disableUI), this, this.disposables);
```

Comment:
We keep our own copy of UIDisplayOptions here.
This is because this class is called in another place to pre-warn the jupyter notebook server (non-raw).
Hence its possible we pre-warm first, and later we start a kernel (i.e. this is called from two different palces with two different object instances for UIDisplayOptions).